execution_engine/src/core/runtime/mod.rs

```diff
@@ -1822,7 +1818,10 @@ where
         };
 
         // Charge for the gas spent during execution in an isolated runtime.
-        self.gas(runtime.gas_counter() - gas_counter)?;
+        // Assumes that gas spent does not exceed u64::MAX which as all system contract costs are
+        // u32.
```

Comment:
Does it assume that? We're calling `charge_gas_large`, after all?
(On the other hand it probably _would_ be safe to assume it's small, and to call `charge_gas`?)